src/OSPSuite.Presentation/DTO/DimensionSelectionDTO.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Infrastructure.Import.Core;
+
+namespace OSPSuite.Presentation.DTO
+{
+   public class DimensionSelectionDTO
+   {
+      private readonly string _sheetName;
+      private readonly string _description;
+
+      public DimensionSelectionDTO(string sheetName, IReadOnlyList<string> description, ColumnInfo columnInfo, IReadOnlyList<IDimension> supportingDimensions)
+      {
+         _sheetName = sheetName;
+         _description = string.Join(" - ", description);
+         Dimensions = supportingDimensions;
+         Column = columnInfo;
+
+         if(supportingDimensions.Count == 1)
+            SelectedDimension = supportingDimensions.Single();
+      }
+
+      public ColumnInfo Column { get; }
+
+      public IReadOnlyList<IDimension> Dimensions { get; }
+      public IDimension SelectedDimension { get; set; }
+
+      public string DisplayName
+      {
+         get
+         {
+            if (string.IsNullOrEmpty(_description))
+               return $"{_sheetName} - {Column.Name}";
+            return $"{_sheetName} - {Column.Name} - {_description}";
+         }
+      }
+   }
```

Comment:
for CSV files, sheet name will be empty. I assume the name should be then just `{Column.Name} - {_description}`

Reply:
Yes, I refactored it now to deal with csv files better